### ISSUE TITLE
Style/IfUnlessModifier not reporting lines longer than MaxLineLength

### ISSUE THREAD

**Issue by reitermarkus** [rubocop-hq/rubocop#3534]
RuboCop auto-corrects `if/unless`-statements to use the modifier form if the line length after the correction will be less than `MaxLineLength`, but it doesn't actually work in the opposite direction, i.e. if the line is already in modifier form, but is longer than `MaxLineLength`.

---
## Expected behavior

Lines that are using the modifier form which are longer than `MaxLineLength` are reported as offenses, and auto-corrected if specified.
## Actual behavior

Lines that are using the modifier form which are longer than `MaxLineLength` are not reported as offenses, and hence, not auto-corrected.
## Steps to reproduce the problem

Create a file which has line longer than `MaxLineLength` using the modifier form.
## RuboCop version

```
$ rubocop -V
0.43.0 (using Parser 2.3.1.4, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by Drenmi** [comment]
I would venture to guess the reason for this is that the offense would actually be picked up in `Style/MaxLineLength`, and adding auto-correct to that cop would be more or less impossible, because it needs to take into consideration the configuration of all other cops in order to produce the right autocorrect.

Auto-correcting a `MaxLineLength` violation in another cop seems a bit dubious to me, at least at first glance, although I admit I've never given that option much consideration. 😀

**Comment by jonas054** [comment]
I've tried to think of situations that could become messy if `Style/IfUnlessModifier` and `Metrics/LineLength` fight over the same offense, but I can't come up with anything solid. I think it would be OK for `Style/IfUnlessModifier` to auto-correct modifler forms into the normal form if the line is longer than `MaxLineLength`.

**Comment by MikeMcQuaid** [comment]
Homebrew would still find this useful so just commenting so stale bot knows someone cares.

**Comment by jfelchner** [comment]
@jonas054 I believe this line length check needs to be an option unless you can literally ask the `LineLength` cop if the line is valid.  The `LineLength` cop allows for things such as regex patterns to ignore, etc which means that just checking for the `MaxLineLength` is insufficient.  The `rubocop:disable` case above is just one edge case of a possible multitude.

**Comment by jonas054** [comment]
You're right, @jfelchner. I'd be tempted to move the check on modifier lines into the `LineLength`cop if it wasn't for the fact that the updated `IfUnlessModifier` has been released, and users may have started to disable this offense locally because of the false positives in edge cases.

I'll try to fix it so that `IfUnlessModifier` asks `LineLength` if the line is too long. Probably through some shared code.
